lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -133,6 +133,43 @@ void sendObjectPostDelete(const Ufe::SceneItem::Ptr& sceneItem)
 #endif
 }
 
+void sendObjectDestroyed(const Ufe::Path& ufePath)
+{
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    Ufe::Scene::instance().notify(Ufe::ObjectDestroyed(ufePath));
+#else
+    // Unfortunately in Ufe v1 there was no object destroyed notif
+    // and the only delete notifs we have both take a scene item
+    // which we cannot create for the input Ufe path since that
+    // path is no longer valid (it has already been destroyed).
+    // So the only choice we have is to subtree invalidate the
+    // parent which will remove the destroyed item (keeping all
+    // the valid children).
+    auto sceneItem = Ufe::Hierarchy::createItem(ufePath);
```

Comment:
As discussed, if we expect this to fail (because the UFE path is to a destroyed object), then we should skip directly to line 151 (creating the parent path), and TF_VERIFY'ing that this succeeds in the conditional at line 154.